### PR TITLE
CVE-2009-2901

### PR DESCRIPTION
**Pull request by based-avocado**
Documents and analyzes the fix related to CVE-2009-2901 within its corresponding `.yml` file.

### CODE REVIEW

File: cves/CVE-2009-2901.yml [hunk 1]
```diff
@@ -4,33 +4,33 @@ nickname_instructions: |
...
   alongside its CVE to make it more easily distinguished from the rest.
-nickname:
+nickname: Insecure partial deploy after failed undeploy
 
 CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE:
+CWE: CWE-212 - Improper Cross-boundary Removal of Sensitive Data
```
Comment (renchauret) [inline cves/CVE-2009-2901.yml]: This should only be the integer "212"

File: cves/CVE-2009-2901.yml [hunk 2]
```diff
@@ -139,7 +152,11 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this
     section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    Not really. There were a handful of bugfixes, a lot of updates to STATUS.txt
+    (which seems to be just tracking progress of existing bug fixes), but no real
+    changes to anything that would touch deployment. This bug was likely in the
+    system for a good while.
```
Comment (renchauret) [inline cves/CVE-2009-2901.yml]: Perhaps rephrase last sentence to be less conversational

File: cves/CVE-2009-2901.yml [hunk 3]
```diff
@@ -202,4 +223,10 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
+    From the looks of it, this is just a good ol' fashion coding mistake. What
```
Comment (renchauret) [inline cves/CVE-2009-2901.yml]: Again, maybe want to be less conversational, but that's just personal preference

**Comment by renchauret** [comment]
1 upvote

File: cves/CVE-2009-2901.yml (hunk 1 above)
Comment (randallkentbrown) [reply cves/CVE-2009-2901.yml]: Agree with Ren.

File: cves/CVE-2009-2901.yml [hunk 4]
```diff
@@ -116,10 +127,12 @@ discovered:
     If there is no evidence as to how this vulnerability was found, then you
     may leave the entries blank except for "answer", BUT please write down
     where you looked in "answer".
-  answer:
-  date:
-  automated:
-  contest:
+  answer: |
+    This find is credited to the Apache Tomcat security team, as seen here -
```
Comment (randallkentbrown) [inline cves/CVE-2009-2901.yml:132]: I assume this means there is no single person called out specifically?

File: cves/CVE-2009-2901.yml [hunk 5]
```diff
@@ -183,8 +200,12 @@ lessons:
     applies:
     note:
   secure_by_default:
-    applies:
-    note:
+    applies: true
+    note: |
+      Part of the reason that this was bug occurred was because of the "autoDeploy
+      attribute of a host defaults to true," which was causing deployment to happen
+      for remaing files following a failed undeploy. These default settings, which
```
Comment (randallkentbrown) [inline cves/CVE-2009-2901.yml]: Typo in "remaing."

File: cves/CVE-2009-2901.yml [hunk 6]
```diff
@@ -43,7 +43,14 @@ description_instructions: |
...
   keep too.
-description:
+description: |
+  By default, Tomcat automatically deploys any directories placed in a host's appBase.
+  This behaviour is controlled by the autoDeploy attribute of a host, which defaults to
+  true. After a failed undeploy, the remaining files will be deployed as a result of
+  the autodeployment process. Depending on circumstances, files normally protected by
+  one or more security constraints may be deployed without those security constraints,
+  making them accessible without authentication. This issue only affects Windows
+  platforms.
```
Comment (randallkentbrown) [inline cves/CVE-2009-2901.yml]: For clarity, the program would attempt to deploy files, fail, then deploy the rest of the files, even after reporting a failure?

File: cves/CVE-2009-2901.yml (hunk 5 above)
Comment (based-avocado) [reply cves/CVE-2009-2901.yml]: Good catch! Will update in a bit.

File: cves/CVE-2009-2901.yml (hunk 1 above)
Comment (based-avocado) [reply cves/CVE-2009-2901.yml]: Sure thing, will fix in a bit.

File: cves/CVE-2009-2901.yml (hunk 2 above)
Comment (based-avocado) [reply cves/CVE-2009-2901.yml]: I generally write in a fairly casual way, which may not necessarily reflect the nature of the assignment. I'll rewrite it to make it sound a little more professional.

File: cves/CVE-2009-2901.yml (hunk 3 above)
Comment (based-avocado) [reply cves/CVE-2009-2901.yml]: Sure, I can make this answer sound a bit more professional too.

File: cves/CVE-2009-2901.yml (hunk 4 above)
Comment (based-avocado) [reply cves/CVE-2009-2901.yml:132]: No credit was given to an individual (that I could find anyways). The best answer I could get for a find was "the Apache Tomcat security team," which is what I opted to put here.

File: cves/CVE-2009-2901.yml (hunk 6 above)
Comment (based-avocado) [reply cves/CVE-2009-2901.yml]: Almost - in this case, the system would attempt to _un_deploy (clean up files), fail, and then deploy the rest of the files in an unsafe way, due to the autodeployment process. I'll update the description to better reflect this.

**Comment by randallkentbrown** [comment]
2 upvotes

**Pull request merged**